### PR TITLE
Build fix for mbedk64f target

### PR DESCRIPTION
**Pull request by robertsipka**
JerryScript-DCO-1.0-Signed-off-by: Robert Sipka rsipka.uszeged@partner.samsung.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
LGTM

File: jerry-core/jerry-port.c [hunk 1]
```diff
@@ -50,5 +50,5 @@ int jerry_port_errormsg (const char *format, /**< format string */
  */
 int jerry_port_putchar (int c) /**< character to put */
 {
-  return putchar (c);
+  return putchar ((unsigned char) c);
```
Comment (zherczeg) [inline jerry-core/jerry-port.c]: Why this is a problem?

http://www.cplusplus.com/reference/cstdio/putchar/

putchar should accept int argument.

**Comment by robertsipka** [comment]
@zherczeg : A conversion error (conversion to 'unsigned char' from 'int' may alter its value) occurs when I tried to build to mbedk64f target with enabled USE_COMPILER_DEFAULT_LIBC. 
It's the default value of this option (hard-coded in the Makefile.mbedk64f), and the same happens when I compile another board with the following options:
USE_COMPILER_DEFAULT_LIBC=YES make release.mcu_stm32f3-cp
As I assume this is a compiler-specific error, and I dared to casting int to unsigned char because the value is internally converted to an unsigned char anyway.

**Comment by zherczeg** [comment]
Btw could we make jerry_port_putchar to accept unsigned char? Is it frequently used?

**Comment by robertsipka** [comment]
No, it's very rarely used, but we couldn't get away with the cast, for example the following line :
jerry_port_putchar (ecma_char)
 will be reported the same error, because the type of the variable is an ecma_char_t which is a typedef to uint16_t.
(In case of this kind of build configuration)

**Comment by zherczeg** [comment]
LGTM

**Comment by robertsipka** [comment]
Rebased with master.

**Pull request merged**